nx/lib/nx/backend.ex

```diff
@@ -234,10 +249,68 @@ defmodule Nx.Backend do
     chunk_each(dim - 1, rest, [doc | acc], limit, fun)
   end
 
-  defp inspect_value(%Complex{} = val), do: Complex.to_string(val)
-  defp inspect_value(integer) when is_integer(integer), do: Integer.to_string(integer)
-  defp inspect_value(float) when is_float(float), do: Float.to_string(float)
-  defp inspect_value(:neg_infinity), do: "-Inf"
-  defp inspect_value(:infinity), do: "Inf"
-  defp inspect_value(:nan), do: "NaN"
+  defp inspect_value(float, precision) when is_float(float) and is_integer(precision) do
+    float_to_string(float, precision)
+  end
+
+  defp inspect_value(float, nil) when is_float(float), do: Float.to_string(float)
+
+  defp inspect_value(%Complex{} = val, precision), do: complex_to_string(val, precision)
+
+  defp inspect_value(integer, _) when is_integer(integer), do: Integer.to_string(integer)
+  defp inspect_value(:neg_infinity, _), do: "-Inf"
+  defp inspect_value(:infinity, _), do: "Inf"
+  defp inspect_value(:nan, _), do: "NaN"
+
+  defp float_to_string(float, precision) do
+    str = :erlang.float_to_binary(float, scientific: precision)
```

Comment:
This actually makes things worse in many scenarios, because it will always enforce an amount of digits, even when shorter representations are available:

```elixir
iex(4)> :erlang.float_to_binary 0.25, [scientific: 6]
"2.500000e-01"
iex(5)> Float.to_string 0.25
"0.25"
```

We would need to prune the result of `Float.to_string` by slightly fixing it afterwards.